common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -69,11 +68,10 @@ public int compare(CharSequence o1, CharSequence o2) {
             int length2 = o2.length();
             int minLength = Math.min(length1, length2);
             if (a1 != null && a2 != null) {
-                byte[] thisValue = a1.value;
-                byte[] thatValue = a2.value;
-                for (int i = 0; i < minLength; i++) {
-                    byte v1 = thisValue[i];
-                    byte v2 = thatValue[i];
+                final int a1Len = minLength + a1.arrayOffset();
```

Comment:
@Scottmitch can you add a benchmark that shows that keeping this large amounts of code is significantly faster than just using a single loop with `charAt`?